qiskit/pulse/transforms.py

```diff
@@ -31,92 +31,103 @@ def align_measures(schedules: Iterable[interfaces.ScheduleComponent],
-    wait time (to allow for calibration pulses) is enforced.
+    """Return new schedules where measurements occur at the same physical time.
+
+    Minimum measurement wait time (to allow for calibration pulses) is enforced
+    and may be set with ```max_calibration_duration`.
 
-    This is only defined for schedules that are acquire-less or acquire-final per channel: a
-    schedule with pulses or acquires occurring on a channel which has already had a measurement will
-    throw an error.
+    By default only instructions containing a :class:`~qiskit.pulse.AcquireChannel`
+    or :class:`~qiskit.pulse.MeasureChannel` will be shifted. If you wish to keep
+    the relative timing of all instructions in the schedule set ``align_all=True``.
+
+    This method assumes that ``MeasureChannel(i)`` and ``AcquireChannel(i)``
+    correspond to the same qubit and the acquire/play instructions
+    should be shifted together on these channels.
 
     Args:
         schedules: Collection of schedules to be aligned together
         inst_map: Mapping of circuit operations to pulse schedules
         cal_gate: The name of the gate to inspect for the calibration time
         max_calibration_duration: If provided, inst_map and cal_gate will be ignored
         align_time: If provided, this will be used as final align time.
-
+        align_all: Shift all instructions such that they happen at the same time
+            relative to the measurement post-alignment. If `False` on
     Returns:
         The input list of schedules transformed to have their measurements aligned.
 
     Raises:
-        PulseError: if an acquire or pulse is encountered on a channel that has already been part
-                    of an acquire, or if align_time is negative
+        PulseError: If the provided alignment time is negative.
     """
-    def calculate_align_time():
-        """Return the the max between the duration of the calibration time and the absolute time
-        of the latest scheduled acquire.
-        """
-        nonlocal max_calibration_duration
-        if max_calibration_duration is None:
-            max_calibration_duration = get_max_calibration_duration()
-        align_time = max_calibration_duration
+    def get_acquire_times(schedules):
+        """Return a list of first acquire times for each schedule."""
+        acquire_times = []
         for schedule in schedules:
-            last_acquire = 0
-            acquire_times = [time for time, inst in schedule.instructions
-                             if isinstance(inst, instructions.Acquire)]
-            if acquire_times:
-                last_acquire = max(acquire_times)
-            align_time = max(align_time, last_acquire)
-        return align_time
-
-    def get_max_calibration_duration():
+            visited_channels = set()
+            sched_acquire_times = defaultdict(lambda: None)
+
+            for time, inst in schedule.instructions:
```

Comment:
We can use filter method with acquire channels.

Reply:
This requires constructing new schedule. I only need to perform analysis of an existing schedule.